src/azure-cli/requirements.py3.windows.txt

```diff
@@ -139,3 +139,4 @@ vsts==0.1.25
 wcwidth==0.1.7
 websocket-client==1.3.1
 xmltodict==0.12.0
+psycopg2-binary==2.9.3
```

Comment:
@xfz11 do you add it in setup.py file of azure-cli https://github.com/Azure/azure-cli/blob/58a8f65c6ec1c098a714107fe7fb2dedda95917f/src/azure-cli/setup.py#L53-L156

Reply:
Why you need the `binary` library?

Reply:
Added in setup.py but build still failed.
If we don't use the `psycopg2-binary`, but install Psycopg from sources, `python3-dev` and `libpq-dev` package are required to build it. Any guide to add those dependencies in CLI build environment?

Reply:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=1575771&view=logs&jobId=aedd744b-2bf4-59d5-bada-b121261b4f81&j=aedd744b-2bf4-59d5-bada-b121261b4f81&t=adba5b8c-e58b-5021-0563-de243d1b6648

```
2022-05-18T07:10:56.2535244Z Provides: azure-cli = 2.36.0-1.el7 azure-cli(x86-64) = 2.36.0-1.el7 config(azure-cli) = 2.36.0-1.el7 libcom_err-2abe824b.so.2.1()(64bit) libcrypto-2f2fcd35.so.1.1()(64bit) libcrypto.so.1.1(OPENSSL_1_1_0)(64bit) libcrypto.so.1.1(OPENSSL_1_1_0a)(64bit) libcrypto.so.1.1(OPENSSL_1_1_0c)(64bit) libcrypto.so.1.1(OPENSSL_1_1_0d)(64bit) libcrypto.so.1.1(OPENSSL_1_1_0f)(64bit) libcrypto.so.1.1(OPENSSL_1_1_0g)(64bit) libcrypto.so.1.1(OPENSSL_1_1_0h)(64bit) libcrypto.so.1.1(OPENSSL_1_1_0i)(64bit) libcrypto.so.1.1(OPENSSL_1_1_0j)(64bit) libcrypto.so.1.1(OPENSSL_1_1_1)(64bit) libcrypto.so.1.1(OPENSSL_1_1_1b)(64bit) libcrypto.so.1.1(OPENSSL_1_1_1c)(64bit) libcrypto.so.1.1(OPENSSL_1_1_1d)(64bit) libcrypto.so.1.1(OPENSSL_1_1_1e)(64bit) libcrypto.so.1.1(OPENSSL_1_1_1h)(64bit) libffi-c643fa1a.so.6.0.4()(64bit) libgssapi_krb5-497db0c6.so.2.2()(64bit) libgssapi_krb5.so.2(HIDDEN)(64bit) libgssapi_krb5.so.2(gssapi_krb5_2_MIT)(64bit) libk5crypto-b1f99d5c.so.3.1()(64bit) libk5crypto.so.3(HIDDEN)(64bit) libk5crypto.so.3(k5crypto_3_MIT)(64bit) libkeyutils-dfe70bd6.so.1.5()(64bit) libkeyutils.so.1(KEYUTILS_0.3)(64bit) libkeyutils.so.1(KEYUTILS_1.0)(64bit) libkeyutils.so.1(KEYUTILS_1.3)(64bit) libkeyutils.so.1(KEYUTILS_1.4)(64bit) libkeyutils.so.1(KEYUTILS_1.5)(64bit) libkrb5-fc820a1d.so.3.3()(64bit) libkrb5.so.3(HIDDEN)(64bit) libkrb5.so.3(krb5_3_MIT)(64bit) libkrb5support-a4e68a84.so.0.1()(64bit) libkrb5support.so.0(HIDDEN)(64bit) libkrb5support.so.0(krb5support_0_MIT)(64bit) liblber-2-f149c062.4.so.2.11.7()(64bit) libldap_r-2-aa060733.4.so.2.11.7()(64bit) libpcre-9513aab5.so.1.2.0()(64bit) libpq-30c1c89d.so.5.14()(64bit) libsasl2-8fd1a564.so.3.0.0()(64bit) libselinux-0922c95c.so.1()(64bit) libssl-6b7b94d5.so.1.1()(64bit) libssl.so.1.1(OPENSSL_1_1_0)(64bit) libssl.so.1.1(OPENSSL_1_1_0d)(64bit) libssl.so.1.1(OPENSSL_1_1_1)(64bit) libssl.so.1.1(OPENSSL_1_1_1a)(64bit)

2022-05-18T07:10:56.2544788Z Requires(rpmlib): rpmlib(CompressedFileNames) <= 3.0.4-1 rpmlib(FileDigests) <= 4.6.0-1 rpmlib(PayloadFilesHavePrefix) <= 4.0-1

2022-05-18T07:10:56.2551092Z Requires: ld-linux-x86-64.so.2()(64bit) ld-linux-x86-64.so.2(GLIBC_2.3)(64bit) libc.so.6()(64bit) libc.so.6(GLIBC_2.12)(64bit) libc.so.6(GLIBC_2.14)(64bit) libc.so.6(GLIBC_2.16)(64bit) libc.so.6(GLIBC_2.17)(64bit) libc.so.6(GLIBC_2.2.5)(64bit) libc.so.6(GLIBC_2.3)(64bit) libc.so.6(GLIBC_2.3.2)(64bit) libc.so.6(GLIBC_2.3.4)(64bit) libc.so.6(GLIBC_2.4)(64bit) libc.so.6(GLIBC_2.6)(64bit) libc.so.6(GLIBC_2.7)(64bit) libc.so.6(GLIBC_2.8)(64bit) libcom_err-2abe824b.so.2.1()(64bit) libcrypto-2f2fcd35.so.1.1()(64bit) libcrypto-2f2fcd35.so.1.1(OPENSSL_1_1_0)(64bit) libcrypto-2f2fcd35.so.1.1(OPENSSL_1_1_0d)(64bit) libcrypto-2f2fcd35.so.1.1(OPENSSL_1_1_0f)(64bit) libcrypto-2f2fcd35.so.1.1(OPENSSL_1_1_0i)(64bit) libcrypto-2f2fcd35.so.1.1(OPENSSL_1_1_1)(64bit) libdl.so.2()(64bit) libdl.so.2(GLIBC_2.2.5)(64bit) libffi-c643fa1a.so.6.0.4()(64bit) libgssapi_krb5-497db0c6.so.2.2()(64bit) libgssapi_krb5-497db0c6.so.2.2(gssapi_krb5_2_MIT)(64bit) libk5crypto-b1f99d5c.so.3.1()(64bit) libk5crypto-b1f99d5c.so.3.1(k5crypto_3_MIT)(64bit) libkeyutils-dfe70bd6.so.1.5()(64bit) libkeyutils-dfe70bd6.so.1.5(KEYUTILS_0.3)(64bit) libkeyutils-dfe70bd6.so.1.5(KEYUTILS_1.0)(64bit) libkeyutils-dfe70bd6.so.1.5(KEYUTILS_1.5)(64bit) libkrb5-fc820a1d.so.3.3()(64bit) libkrb5-fc820a1d.so.3.3(krb5_3_MIT)(64bit) libkrb5support-a4e68a84.so.0.1()(64bit) libkrb5support-a4e68a84.so.0.1(krb5support_0_MIT)(64bit) liblber-2-f149c062.4.so.2.11.7()(64bit) libldap_r-2-aa060733.4.so.2.11.7()(64bit) libpcre-9513aab5.so.1.2.0()(64bit) libpthread.so.0()(64bit) libpthread.so.0(GLIBC_2.2.5)(64bit) libpthread.so.0(GLIBC_2.3.2)(64bit) libresolv.so.2()(64bit) libresolv.so.2(GLIBC_2.2.5)(64bit) libsasl2-8fd1a564.so.3.0.0()(64bit) libselinux-0922c95c.so.1()(64bit) libssl-6b7b94d5.so.1.1()(64bit) libssl-6b7b94d5.so.1.1(OPENSSL_1_1_0)(64bit) libssl-6b7b94d5.so.1.1(OPENSSL_1_1_1)(64bit) libz.so.1()(64bit) rtld(GNU_HASH)
...
Step 13/13 : RUN rpm -i ./azure-cli-dev.rpm &&     az --version
 ---> Running in db374f835afc
error: Failed dependencies:
	libcrypto-2f2fcd35.so.1.1(OPENSSL_1_1_0)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libcrypto-2f2fcd35.so.1.1(OPENSSL_1_1_0d)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libcrypto-2f2fcd35.so.1.1(OPENSSL_1_1_0f)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libcrypto-2f2fcd35.so.1.1(OPENSSL_1_1_0i)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libcrypto-2f2fcd35.so.1.1(OPENSSL_1_1_1)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libgssapi_krb5-497db0c6.so.2.2(gssapi_krb5_2_MIT)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libk5crypto-b1f99d5c.so.3.1(k5crypto_3_MIT)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libkeyutils-dfe70bd6.so.1.5(KEYUTILS_0.3)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libkeyutils-dfe70bd6.so.1.5(KEYUTILS_1.0)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libkeyutils-dfe70bd6.so.1.5(KEYUTILS_1.5)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libkrb5-fc820a1d.so.3.3(krb5_3_MIT)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libkrb5support-a4e68a84.so.0.1(krb5support_0_MIT)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libssl-6b7b94d5.so.1.1(OPENSSL_1_1_0)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
	libssl-6b7b94d5.so.1.1(OPENSSL_1_1_1)(64bit) is needed by azure-cli-2.36.0-1.el7.x86_64
The command '/bin/sh -c rpm -i ./azure-cli-dev.rpm &&     az --version' returned a non-zero code: 1
```

I think there is some problem with `psycopg2-binary` that it requires libs like `libcrypto-2f2fcd35.so.1.1(OPENSSL_1_1_0)(64bit)`, but doesn't provide it, meaning it is not compatible with `rpmbuild` and `rpm`. This could be similar to #20608.

Reply:
We should use `psycopg2` as dependency when build package, which is suggested by [psycopg owner](https://github.com/psycopg/psycopg2/issues/1460). It has some pre-required package, to reduce the complexity, we may use `psycopg2-binary` in develop environment.

Reply:
> We should use `psycopg2` as dependency when build package, which is suggested by [psycopg owner](https://github.com/psycopg/psycopg2/issues/1460). It has some pre-required package, to reduce the complexity, we may use `psycopg2-binary` in develop environment.

Thanks for your help! Since `psycopg2` has other library dependencies on Linux and MacOS. I update the code to dynamically install the packages only when user run the command. For windows, it needs no other library, so I add the package in windows requirement files.